### ISSUE TITLE
Update CKEditor website links

### ISSUE THREAD

**Issue by mlewand** [ckeditor/ckeditor-docs#111]
There are couple of outdated links, e.g. home page has:

* official website - linking to a http://ckeditor.com/license 404
* Developers Community Site linking to http://dev.ckeditor.com/ - this should be removed all along
* report bugs - http://dev.ckeditor.com/ should be changed to our https://github.com/ckeditor/ckeditor-dev/issues

Quick Start Guide has:

* CKEditor Download - should link to https://ckeditor.com/ckeditor-4/download/

Basically to gather links like that you could perform a search using `http://ckeditor.com`, `"ckeditor.com` throughout the repo.

Also whenever possible, please use `https` links.

**Comment by AnnaTomanek** [comment]
I think I fixed everything now, apart from Legal/license (to be discussed). Feel free to reopen if you find anything else.
